odoo/addons/base/models/ir_rule.py

```diff
@@ -230,48 +203,40 @@ def _make_access_error(self, operation, records):
         resolution_info = _("Contact your administrator to request access if necessary.")
 
         if not self.env.user.has_group('base.group_no_one') or not self.env.user.has_group('base.group_user'):
-            msg = """{operation_error}
-
-{resolution_info}""".format(
-                operation_error=operation_error,
-                resolution_info=resolution_info)
-            return AccessError(msg)
+            return AccessError(f"{operation_error}\n\n{resolution_info}")
 
         # This extended AccessError is only displayed in debug mode.
         # Note that by default, public and portal users do not have
         # the group "base.group_no_one", even if debug mode is enabled,
         # so it is relatively safe here to include the list of rules and record names.
         rules = self._get_failing(records, mode=operation).sudo()
 
-        records_description = ', '.join(['%s (id=%s)' % (rec.display_name, rec.id) for rec in records[:6].sudo()])
+        records_sudo = records[:6].sudo()
```

Comment:
While we are at it, do we still need to access these records in sudo? It's a small leak of information than showing the `display_name` of forbidden records.
Example [private address](https://user-images.githubusercontent.com/564822/146003055-f51dadee-df62-43cd-92fd-7207d3c1fe81.png) - [error](https://user-images.githubusercontent.com/564822/146003179-08a1202a-7387-4cf4-b93e-16fe01b6379b.png) (when trying to read it with another user)
I know it's a tradeoff to make with usability.

Reply:
Since the entire point is that the user doen't have access to these records, not sudo-ing them would recursively blow up when accessing the `company_id` and possibly the `display_name` (depending how that resolves), no?

Reply:
The field `display_name` is what is shown for a many2one field in a form view, whether you can access the corresponding record or not.  This is the _purpose_ of `display_name`.  So putting sensitive data in `display_name` looks like a very bad idea to me.

Reply:
Yep as @rco-odoo said, all records display_name is already accessible (`read` + `convert_to_read` of Many2one) from any accessible Many2one fields.
However, how does the `display_name` help in the message to debug ?

Reply:
@ryv-odoo assume you have access to companies A and B.
Now you work in company A, with company B enabled as well.
It means that you can see records linked to both companies, but can only perform actions on records of company A.
If you try to perform an action on a record linked to company B, you may get an error.
And the error message gives you the hint: the record is linked to company B.
With that information, you know that switching to company B may give you the expected access rights to perform the action on the given record.

Reply:
Yep, understood, it seems legit. Let keep it that way.

Reply:
> The field `display_name` is what is shown for a many2one field in a form view, whether you can access the corresponding record or not. This is the _purpose_ of `display_name`. So putting sensitive data in `display_name` looks like a very bad idea to me.

Sure but that doesn't mean all `display_name`/`name_get` are "correct" as this is not enforced, furthermore this can be context-dependent (e.g. innocuous `name_get` in some contexts, less so in others).

In fact maybe it'd be a good idea to clean up the context? Assuming contextual name_gets are "fail-safe" (an empty context is innocuous) we should probably filter the context to keep only the well-known and innocent keys (`lang` and the like)?

`res.partner` has a custom override for this, but we can't necessarily assume that's the case of every model with a wonky `name_get` override.